popde/density_estimate.py

```diff
@@ -74,6 +77,18 @@ def __init__(self, data, input_transf=None, stdize=False, rescale=None,
         if dim_names is not None:
             self.check_dimensionality()
 
+        self.weights = weights
+        if self.weights is not None:
+            #check the array
+            self._weights = atleast_1d(weights).astype(float)
```

Comment:
I am not seeing what this `_weights` attribute is for or why we have both this and self.weights.

Reply:
fixed it.